### PR TITLE
feat: curio: Storage reservations in MoveStorage

### PR DESCRIPTION
**Pull request by magik6k**
## Proposed Changes
* Plumb storage reservation logic into MoveStorage
* Add Storage to the MoveStorage task

## Additional Info
Todo:
* [x] Fisish impl
* [x] Test on mainnet

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: curiosrc/ffi/sdr_funcs.go [hunk 1]
```diff
@@ -489,7 +489,24 @@ func (sb *SealCalls) MoveStorage(ctx context.Context, sector storiface.SectorRef
 
 	toMove := storiface.FTCache | storiface.FTSealed | moveUnsealed
 
-	err := sb.sectors.storage.MoveStorage(ctx, sector, toMove)
+	var opts []storiface.AcquireOption
+	if taskID != nil {
+		resv, ok := sb.sectors.storageReservations.Load(*taskID)
+		if ok {
```
Comment (snadrus) [inline curiosrc/ffi/sdr_funcs.go:495]: what if !ok ?

Comment (magik6k) [reply curiosrc/ffi/sdr_funcs.go:495]: Then everything still works, just without the storage reservation - which iiuc can only happen on restart, and the reservation will get restored as soon as the task starts (MoveStorage will reserve space internally if not provided with a reservation).

Comment (snadrus) [reply curiosrc/ffi/sdr_funcs.go:495]: A comment here would be helpful

File: curiosrc/seal/task_movestorage.go [hunk 2]
```diff
@@ -136,18 +136,39 @@ func (m *MoveStorageTask) CanAccept(ids []harmonytask.TaskID, engine *harmonytas
...
 		Max:  m.max,
 		Name: "MoveStorage",
 		Cost: resources.Resources{
-			Cpu: 1,
-			Gpu: 0,
-			Ram: 128 << 20,
+			Cpu:     1,
+			Gpu:     0,
+			Ram:     128 << 20,
+			Storage: m.sc.Storage(m.taskToSector, storiface.FTNone, storiface.FTCache|storiface.FTSealed|storiface.FTUnsealed, ssize, storiface.PathStorage),
```
Comment (snadrus) [inline curiosrc/seal/task_movestorage.go:151]: Can this change to read ssize in dynamically so we request the correct size?

**Review by snadrus — APPROVED** [review]

File: curiosrc/seal/task_movestorage.go (hunk 2 above)
Comment (magik6k) [reply curiosrc/seal/task_movestorage.go:151]: This is plumbed to HasCapacity so not really sadly.

**Pull request merged**